src/Http/Controller/SearchController.php

```diff
@@ -7,18 +7,50 @@
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Annotation\Route;
 
+use App\Domain\Course\Entity\Technology;
+use App\Http\Normalizer\TechnologyPathNormalizer;
+
+
 class SearchController extends AbstractController
 {
     /**
      * @Route("/search", name="search")
      */
-    public function search(Request $request, SearchInterface $search): Response
-    {
+    public function search(
+      Request $request,
+      SearchInterface $search,
+      TechnologyPathNormalizer $technologyPathNormalizer
+    ): Response {
         $q = $request->query->get('q');
 
-        return $this->render('pages/search.html.twig', [
-            'q' => $q,
-            'results' => $search->search($q, [])['hits'],
-        ]);
+        // Si q n'est pas vide
+        if ( !empty( $q ) ) {
+
+          // on commence par chercher une techno correspondante
+          $technology = $this->getDoctrine()
+              ->getRepository(Technology::class)
+              ->findOneBy( ['name' => $q] );
+
+          // si ça matche on fait la redirection vers la techno en normalisant le path
+          if ( !empty( $technology ) ) {
+            return $this->redirectToRoute(
+              $technologyPathNormalizer->normalize( $technology )['path'],
+              $technologyPathNormalizer->normalize( $technology )['params']
+            );
```

Comment:
Tu peux stocker la normalisation de la technologie pour éviter de le faire deux fois.
```suggestion
            $path = $technologyPathNormalizer->normalize($technology);
            return $this->redirectToRoute($path['path'], $path['params']);
```